core/field.js

```diff
@@ -64,1155 +64,1194 @@ goog.require('Blockly.Gesture');
+     * @type {SVGGElement}
+     * @protected
+     */
+    this.fieldGroup_ = null;
+
+    /**
+     * The rendered field's SVG border element.
+     * @type {SVGRectElement}
+     * @protected
+     */
+    this.borderRect_ = null;
+
+    /**
+     * The rendered field's SVG text element.
+     * @type {SVGTextElement}
+     * @protected
+     */
+    this.textElement_ = null;
+
+    /**
+     * The rendered field's text content element.
+     * @type {Text}
+     * @protected
+     */
+    this.textContent_ = null;
+
+    /**
+     * Mouse down event listener data.
+     * @type {?browserEvents.Data}
+     * @private
+     */
+    this.mouseDownWrapper_ = null;
+
+    /**
+     * Constants associated with the source block's renderer.
+     * @type {ConstantProvider}
+     * @protected
+     */
+    this.constants_ = null;
+
+    /**
+     * Has this field been disposed of?
+     * @type {boolean}
+     * @package
+     */
+    this.disposed = false;
+
+    /**
+     * Maximum characters of text to display before adding an ellipsis.
+     * @type {number}
+     */
+    this.maxDisplayLength = 50;
+
+    /**
+     * Block this field is attached to.  Starts as null, then set in init.
+     * @type {Block}
+     * @protected
+     */
+    this.sourceBlock_ = null;
+
+    /**
+     * Does this block need to be re-rendered?
+     * @type {boolean}
+     * @protected
+     */
+    this.isDirty_ = true;
+
+    /**
+     * Is the field visible, or hidden due to the block being collapsed?
+     * @type {boolean}
+     * @protected
+     */
+    this.visible_ = true;
+
+    /**
+     * Can the field value be changed using the editor on an editable block?
+     * @type {boolean}
+     * @protected
+     */
+    this.enabled_ = true;
+
+    /**
+     * The element the click handler is bound to.
+     * @type {Element}
+     * @protected
+     */
+    this.clickTarget_ = null;
+
+    /**
+     * The prefix field.
+     * @type {?string}
+     * @package
+     */
+    this.prefixField = null;
+
+    /**
+     * The suffix field.
+     * @type {?string}
+     * @package
+     */
+    this.suffixField = null;
+
+    /**
+     * Editable fields usually show some sort of UI indicating they are
+     * editable. They will also be saved by the serializer.
+     * @type {boolean}
+     */
+    this.EDITABLE = true;
+
+    /**
+     * Serializable fields are saved by the serializer, non-serializable fields
+     * are not. Editable fields should also be serializable. This is not the
+     * case by default so that SERIALIZABLE is backwards compatible.
+     * @type {boolean}
+     */
+    this.SERIALIZABLE = false;
+
+    /**
+     * Mouse cursor style when over the hotspot that initiates the editor.
+     * @type {string}
+     */
+    this.CURSOR = '';
+
+    if (value == Field.SENTINEL) return;
+    if (opt_config) this.configure_(opt_config);
+    this.setValue(value);
+    if (opt_validator) this.setValidator(opt_validator);
```

Comment:
So wrt to the config, I think we definitely *don't* want to apply that because it will trigger the same problems that setting the value does. Namely properties getting set by subclasses before they have been initialized in that subclass' constructor.

Then wrt to the validator, I don't think we want to apply that until after we set the initial value, because we don't want it to be triggered until after the field is attached to a block. This is because many external devs create validators that rely on other state in the workspace. So they expect to have a reference to `this.getSourceBlock()` whenever their validator is called. If we break that implicit promise, it will probably be pretty bad.

Now: wrt the question of how to communicate that we're ignoring all of the parameters, I am not sure.